src/main/java/com/echobox/cache/CacheWithSupplierFailover.java

```diff
@@ -66,15 +70,40 @@ public CacheWithSupplierFailover(CacheService cacheService, TypeToken<T> returnT
    * @return cached data
    */
   public T getWithFailover(String key, Supplier<T> sourceOfTruthSupplier) {
-    throw new NotImplementedException("Coming soon");
+    // Only use the supplier once every period, otherwise use cached value
+    String defaultKey = "default-" + key;
+    String errorKey = "error-" + key;
+    T value = cacheService.tryGetCachedItem(defaultKey, returnType);
+    if (Objects.isNull(value)) {
+      try {
+        value = sourceOfTruthSupplier.get();
+        cacheData(defaultKey, value, defaultCacheSecs);
+        cacheData(errorKey, value, maxCacheSecsOnError);
+      } catch (Exception exception) {
+        // Failed to get the value from source of truth. Use cached value
+        logger.warn("Failed to get the latest value for " + defaultKey
+            + ". Trying to get the cached value");
+        value = cacheService.tryGetCachedItem(errorKey, returnType);
+        if (Objects.isNull(value)) {
+          // We exceeded maximum failover interval
+          String message = String.format("The maximum failover timeout of %d seconds has expired.",
```

Comment:
```suggestion
          String message = String.format("The maximum failover timeout of %d seconds has expired for key " + key,
```
Or similar giving your using the string.format.